[Pytorch upstream] Inductor UT raise PassManager::run failed on LTS2 driver.

### Describe the bug

Hi, team: we're updating the pytorch CI driver from LTS to LTS2, and find an Inductor UT regression error: 
RuntimeError: PassManager::run failed.

The error log:
```
torch._inductor.compile_worker.subproc_pool.SubprocException: An exception occurred in a subprocess:

Traceback (most recent call last):
  File "/home/xinanlin/xinanlin/pytorch/torch/_inductor/compile_worker/subproc_pool.py", line 375, in do_job
    result = job()
  File "/home/xinanlin/xinanlin/pytorch/torch/_inductor/runtime/compile_tasks.py", line 62, in _worker_compile_triton
    kernel.precompile(warm_cache_only=True)
  File "/home/xinanlin/xinanlin/pytorch/torch/_inductor/runtime/triton_heuristics.py", line 428, in precompile
    self._precompile_worker()
  File "/home/xinanlin/xinanlin/pytorch/torch/_inductor/runtime/triton_heuristics.py", line 459, in _precompile_worker
    compile_results.append(self._precompile_config(c))
  File "/home/xinanlin/xinanlin/pytorch/torch/_inductor/runtime/triton_heuristics.py", line 748, in _precompile_config
    binary = triton.compile(*compile_args, **compile_kwargs)
  File "/home/xinanlin/xinanlin/miniforge3/lib/python3.10/site-packages/triton/compiler/compiler.py", line 362, in compile
    next_module = compile_ir(module, metadata)
  File "/home/xinanlin/xinanlin/miniforge3/lib/python3.10/site-packages/triton/backends/intel/compiler.py", line 472, in <lambda>
    stages["ttgir"] = lambda src, metadata: self.make_ttgir(src, metadata, options, self.properties)
  File "/home/xinanlin/xinanlin/miniforge3/lib/python3.10/site-packages/triton/backends/intel/compiler.py", line 322, in make_ttgir
    pm.run(mod)
RuntimeError: PassManager::run failed
```
The reproduce script:
```
# AOT ID: ['0_forward']
from ctypes import c_void_p, c_long, c_int
import torch
import math
import random
import os
import tempfile
from math import inf, nan
from cmath import nanj
from torch._inductor.hooks import run_intermediate_hooks
from torch._inductor.utils import maybe_profile
from torch._inductor.codegen.memory_planning import _align as align
from torch import device, empty_strided
from torch._inductor.async_compile import AsyncCompile
from torch._inductor.select_algorithm import extern_kernels
import triton
import triton.language as tl
from torch._inductor.runtime.triton_heuristics import start_graph, end_graph
from torch._C import _xpu_getCurrentRawStream as get_raw_stream
from torch._C import _xpu_getCurrentRawStream as get_raw_stream

aten = torch.ops.aten
inductor_ops = torch.ops.inductor
_quantized = torch.ops._quantized
assert_size_stride = torch._C._dynamo.guards.assert_size_stride
assert_alignment = torch._C._dynamo.guards.assert_alignment
empty_strided_cpu = torch._C._dynamo.guards._empty_strided_cpu
empty_strided_cuda = torch._C._dynamo.guards._empty_strided_cuda
empty_strided_xpu = torch._C._dynamo.guards._empty_strided_xpu
reinterpret_tensor = torch._C._dynamo.guards._reinterpret_tensor
alloc_from_pool = torch.ops.inductor._alloc_from_pool
async_compile = AsyncCompile()
empty_strided_p2p = torch._C._distributed_c10d._SymmetricMemory.empty_strided_p2p


# kernel path: /tmp/torchinductor_xinanlin/4m/c4mlnxbuieiasgmzyvkey7lb5bbc4estajl3lgyqlvb56q6zdpgm.py
# Topologically Sorted Source Nodes: [unfold], Original ATen: [aten.im2col]
# Source node to ATen node mapping:
#   unfold => add, iota, iota_1, unsqueeze, unsqueeze_1
# Graph fragment:
#   %iota : [num_users=1] = call_function[target=torch.ops.prims.iota.default](args = (4,), kwargs = {start: 0, step: 1, dtype: torch.int64, device: xpu:0, requires_grad: False})
#   %unsqueeze : [num_users=1] = call_function[target=torch.ops.aten.unsqueeze.default](args = (%iota, 0), kwargs = {})
#   %iota_1 : [num_users=1] = call_function[target=torch.ops.prims.iota.default](args = (2,), kwargs = {start: 0, step: 1, dtype: torch.int64, device: xpu:0, requires_grad: False})
#   %unsqueeze_1 : [num_users=1] = call_function[target=torch.ops.aten.unsqueeze.default](args = (%iota_1, -1), kwargs = {})
#   %add : [num_users=3] = call_function[target=torch.ops.aten.add.Tensor](args = (%unsqueeze, %unsqueeze_1), kwargs = {})
triton_poi_fused_im2col_0 = async_compile.triton('triton_poi_fused_im2col_0', '''
import triton
import triton.language as tl

from torch._inductor.runtime import triton_helpers, triton_heuristics
from torch._inductor.runtime.triton_helpers import libdevice, math as tl_math
from torch._inductor.runtime.hints import AutotuneHint, ReductionHint, TileHint, DeviceProperties
triton_helpers.set_driver_to_gpu()

@triton_heuristics.pointwise(
    size_hints={'x': 8}, 
    filename=__file__,
    triton_meta={'signature': {'out_ptr0': '*i64', 'xnumel': 'i32', 'XBLOCK': 'constexpr'}, 'device': DeviceProperties(type='xpu', index=0, multi_processor_count=64, cc={'architecture': 13136561920, 'device_id': 3029, 'driver_version': '1.6.33276+21', 'gpu_eu_count': 512, 'gpu_subslice_count': 64, 'has_atomic64': True, 'has_bfloat16_conversions': True, 'has_fp16': True, 'has_fp64': True, 'has_subgroup_2d_block_io': True, 'has_subgroup_matrix_multiply_accumulate': True, 'has_subgroup_matrix_multiply_accumulate_tensor_float32': False, 'max_compute_units': 512, 'max_num_sub_groups': 64, 'max_work_group_size': 1024, 'name': 'Intel(R) Data Center GPU Max 1550', 'platform_name': 'Intel(R) oneAPI Unified Runtime over Level-Zero', 'sub_group_sizes': [16, 32], 'total_memory': 68702699520, 'type': 'gpu', 'vendor': 'Intel(R) Corporation', 'version': '12.60.7'}, major=None, regs_per_multiprocessor=None, max_threads_per_multi_processor=None, warp_size=32), 'constants': {}, 'configs': [{(0,): [['tt.divisibility', 16]]}]},
    inductor_meta={'grid_type': 'Grid1D', 'autotune_hints': set(), 'kernel_name': 'triton_poi_fused_im2col_0', 'mutated_arg_names': [], 'optimize_mem': False, 'no_x_dim': False, 'num_load': 0, 'num_reduction': 0, 'backend_hash': 'BA7EA12C17D260087807A4E601604F3CC736F0F8D0AE38760F7D1B97A7504774', 'are_deterministic_algorithms_enabled': False, 'assert_indirect_indexing': True, 'autotune_local_cache': True, 'autotune_pointwise': True, 'autotune_remote_cache': None, 'force_disable_caches': False, 'dynamic_scale_rblock': True, 'max_autotune': False, 'max_autotune_pointwise': False, 'min_split_scan_rblock': 256, 'spill_threshold': 16, 'store_cubin': False, 'tiling_scores': {'x': 128}},
    min_elem_per_thread=0
)
@triton.jit
def triton_poi_fused_im2col_0(out_ptr0, xnumel, XBLOCK : tl.constexpr):
    xnumel = 8
    xoffset = tl.program_id(0) * XBLOCK
    xindex = xoffset + tl.arange(0, XBLOCK)[:]
    xmask = xindex < xnumel
    x0 = (xindex % 4)
    x1 = xindex // 4
    x2 = xindex
    tmp0 = x0 + x1
    tl.store(tl.make_block_ptr(out_ptr0, shape=[8], strides=[1], block_shape=[XBLOCK], order=[0], offsets=[xoffset]), tl.broadcast_to(tmp0, [XBLOCK]).to(tl.int64), boundary_check=[0])
''', device_str='xpu')


# kernel path: /tmp/torchinductor_xinanlin/ke/ckezo5lp275omoh5gyhtmuqbk7txr3j56bdd67qrripatlqbpr7f.py
# Topologically Sorted Source Nodes: [unfold], Original ATen: [aten.im2col]
# Source node to ATen node mapping:
#   unfold => clone, index, permute
# Graph fragment:
#   %index : [num_users=1] = call_function[target=torch.ops.aten.index.Tensor](args = (%primals_1, [None, None, %unsqueeze_5, %add]), kwargs = {})
#   %permute : [num_users=1] = call_function[target=torch.ops.aten.permute.default](args = (%index, [0, 1, 2, 4, 3, 5]), kwargs = {})
#   %clone : [num_users=1] = call_function[target=torch.ops.aten.clone.default](args = (%permute,), kwargs = {memory_format: torch.contiguous_format})
triton_poi_fused_im2col_1 = async_compile.triton('triton_poi_fused_im2col_1', '''
import triton
import triton.language as tl

from torch._inductor.runtime import triton_helpers, triton_heuristics
from torch._inductor.runtime.triton_helpers import libdevice, math as tl_math
from torch._inductor.runtime.hints import AutotuneHint, ReductionHint, TileHint, DeviceProperties
triton_helpers.set_driver_to_gpu()

@triton_heuristics.pointwise(
    size_hints={'x': 512}, 
    filename=__file__,
    triton_meta={'signature': {'in_ptr0': '*i64', 'in_ptr1': '*fp16', 'out_ptr0': '*fp16', 'xnumel': 'i32', 'XBLOCK': 'constexpr'}, 'device': DeviceProperties(type='xpu', index=0, multi_processor_count=64, cc={'architecture': 13136561920, 'device_id': 3029, 'driver_version': '1.6.33276+21', 'gpu_eu_count': 512, 'gpu_subslice_count': 64, 'has_atomic64': True, 'has_bfloat16_conversions': True, 'has_fp16': True, 'has_fp64': True, 'has_subgroup_2d_block_io': True, 'has_subgroup_matrix_multiply_accumulate': True, 'has_subgroup_matrix_multiply_accumulate_tensor_float32': False, 'max_compute_units': 512, 'max_num_sub_groups': 64, 'max_work_group_size': 1024, 'name': 'Intel(R) Data Center GPU Max 1550', 'platform_name': 'Intel(R) oneAPI Unified Runtime over Level-Zero', 'sub_group_sizes': [16, 32], 'total_memory': 68702699520, 'type': 'gpu', 'vendor': 'Intel(R) Corporation', 'version': '12.60.7'}, major=None, regs_per_multiprocessor=None, max_threads_per_multi_processor=None, warp_size=32), 'constants': {}, 'configs': [{(0,): [['tt.divisibility', 16]], (1,): [['tt.divisibility', 16]], (2,): [['tt.divisibility', 16]], (3,): [['tt.divisibility', 16]]}]},
    inductor_meta={'grid_type': 'Grid1D', 'autotune_hints': set(), 'kernel_name': 'triton_poi_fused_im2col_1', 'mutated_arg_names': [], 'optimize_mem': False, 'no_x_dim': False, 'num_load': 2, 'num_reduction': 0, 'backend_hash': 'BA7EA12C17D260087807A4E601604F3CC736F0F8D0AE38760F7D1B97A7504774', 'are_deterministic_algorithms_enabled': False, 'assert_indirect_indexing': True, 'autotune_local_cache': True, 'autotune_pointwise': True, 'autotune_remote_cache': None, 'force_disable_caches': False, 'dynamic_scale_rblock': True, 'max_autotune': False, 'max_autotune_pointwise': False, 'min_split_scan_rblock': 256, 'spill_threshold': 16, 'store_cubin': False},
    min_elem_per_thread=0
)
@triton.jit
def triton_poi_fused_im2col_1(in_ptr0, in_ptr1, out_ptr0, xnumel, XBLOCK : tl.constexpr):
    xnumel = 384
    xoffset = tl.program_id(0) * XBLOCK
    xindex = xoffset + tl.arange(0, XBLOCK)[:]
    xmask = xindex < xnumel
    x1 = ((xindex // 4) % 4)
    x3 = ((xindex // 32) % 2)
    x0 = (xindex % 4)
    x2 = ((xindex // 16) % 2)
    x4 = xindex // 64
    x7 = xindex
    tmp0 = tl.reshape(tl.broadcast_to(tl.load(tl.make_block_ptr(in_ptr0, shape=[2, 4], strides=[4, 1], block_shape=[((2) * ((2) <= ((31 + XBLOCK) // 32)) + ((31 + XBLOCK) // 32) * (((31 + XBLOCK) // 32) < (2))), ((4) * ((4) <= ((3 + XBLOCK) // 4)) + ((3 + XBLOCK) // 4) * (((3 + XBLOCK) // 4) < (4)))], order=[1, 0], offsets=[((xoffset // 32) % 2), ((xoffset // 4) % 4)]), eviction_policy='evict_last')[None, :, None, :, None], [(63 + XBLOCK) // 64, ((2) * ((2) <= ((31 + XBLOCK) // 32)) + ((31 + XBLOCK) // 32) * (((31 + XBLOCK) // 32) < (2))), ((2) * ((2) <= ((15 + XBLOCK) // 16)) + ((15 + XBLOCK) // 16) * (((15 + XBLOCK) // 16) < (2))), ((4) * ((4) <= ((3 + XBLOCK) // 4)) + ((3 + XBLOCK) // 4) * (((3 + XBLOCK) // 4) < (4))), ((4) * ((4) <= (XBLOCK)) + (XBLOCK) * ((XBLOCK) < (4)))]), [XBLOCK])
    tmp6 = tl.reshape(tl.broadcast_to(tl.load(tl.make_block_ptr(in_ptr0, shape=[2, 4], strides=[4, 1], block_shape=[((2) * ((2) <= ((15 + XBLOCK) // 16)) + ((15 + XBLOCK) // 16) * (((15 + XBLOCK) // 16) < (2))), ((4) * ((4) <= (XBLOCK)) + (XBLOCK) * ((XBLOCK) < (4)))], order=[1, 0], offsets=[((xoffset // 16) % 2), (xoffset % 4)]), eviction_policy='evict_last')[None, :, None, :], [(31 + XBLOCK) // 32, ((2) * ((2) <= ((15 + XBLOCK) // 16)) + ((15 + XBLOCK) // 16) * (((15 + XBLOCK) // 16) < (2))), ((4) * ((4) <= ((3 + XBLOCK) // 4)) + ((3 + XBLOCK) // 4) * (((3 + XBLOCK) // 4) < (4))), ((4) * ((4) <= (XBLOCK)) + (XBLOCK) * ((XBLOCK) < (4)))]), [XBLOCK])
    tmp1 = tl.full([XBLOCK], 5, tl.int32)
    tmp2 = tmp0 + tmp1
    tmp3 = tmp0 < 0
    tmp4 = tl.where(tmp3, tmp2, tmp0)
    tl.device_assert(((0 <= tmp4) & (tmp4 < 5)) | ~(xmask), "index out of bounds: 0 <= tmp4 < 5")
    tmp7 = tmp6 + tmp1
    tmp8 = tmp6 < 0
    tmp9 = tl.where(tmp8, tmp7, tmp6)
    tl.device_assert(((0 <= tmp9) & (tmp9 < 5)) | ~(xmask), "index out of bounds: 0 <= tmp9 < 5")
    tmp11 = tl.load(in_ptr1 + (tmp9 + 5*tmp4 + 25*x4), xmask, eviction_policy='evict_last').to(tl.float32)
    tl.store(tl.make_block_ptr(out_ptr0, shape=[384], strides=[1], block_shape=[XBLOCK], order=[0], offsets=[xoffset]), tl.broadcast_to(tmp11, [XBLOCK]).to(tl.float16), boundary_check=[0])
''', device_str='xpu')


async_compile.wait(globals())
del async_compile

def call(args):
    primals_1, = args
    args.clear()
    assert_size_stride(primals_1, (2, 3, 5, 5), (75, 25, 5, 1))
    with torch.xpu._DeviceGuard(0):
        torch.xpu.set_device(0)
        buf0 = empty_strided_xpu((2, 4), (4, 1), torch.int64)
        # Topologically Sorted Source Nodes: [unfold], Original ATen: [aten.im2col]
        stream0 = get_raw_stream(0)
        triton_poi_fused_im2col_0.run(buf0, 8, stream=stream0)
        buf1 = empty_strided_xpu((2, 3, 2, 2, 4, 4), (192, 64, 32, 16, 4, 1), torch.float16)
        # Topologically Sorted Source Nodes: [unfold], Original ATen: [aten.im2col]
        stream0 = get_raw_stream(0)
        triton_poi_fused_im2col_1.run(buf0, primals_1, buf1, 384, stream=stream0)
        del primals_1
    return (reinterpret_tensor(buf1, (2, 12, 16), (192, 16, 1), 0), buf0, )


def benchmark_compiled_module(times=10, repeat=10):
    from torch._dynamo.testing import rand_strided
    from torch._inductor.utils import print_performance
    primals_1 = rand_strided((2, 3, 5, 5), (75, 25, 5, 1), device='xpu:0', dtype=torch.float16)
    fn = lambda: call([primals_1])
    return print_performance(fn, times=times, repeat=repeat)


if __name__ == "__main__":
    from torch._inductor.wrapper_benchmark import compiled_module_main
    compiled_module_main('None', benchmark_compiled_module)

```

### Environment details

Triton: release 3.4.x
Pytorch: nightly/main branch.
GPU: PVC
OS: Linux

<details>
<summary> Run with TORCH_COMPILE_DEBUG=1 to get more info </summary>

```bash
TORCH_COMPILE_DEBUG=1 python issue_4725.py 
module attributes {"ttg.threads-per-warp" = 32 : i32, ttig.min_sg_size = 16 : i32, ttig.support_bf16_conversion, ttig.support_dpas, ttig.support_sg_2d_block, ttig.target_arch = "spir64"} {
  tt.func public @triton_poi_fused_im2col_1(%arg0: !tt.ptr<i64> {tt.divisibility = 16 : i32}, %arg1: !tt.ptr<f16> {tt.divisibility = 16 : i32}, %arg2: !tt.ptr<f16> {tt.divisibility = 16 : i32}, %arg3: i32 {tt.divisibility = 16 : i32}) attributes {noinline = false} {
    %cst = arith.constant dense<0> : tensor<256xi64>
    %c384_i64 = arith.constant 384 : i64
    %cst_0 = arith.constant dense<25> : tensor<256xi32>
    %cst_1 = arith.constant dense<5> : tensor<256xi64>
    %cst_2 = arith.constant dense<true> : tensor<256xi1>
    %c1_i64 = arith.constant 1 : i64
    %c4_i64 = arith.constant 4 : i64
    %c2_i64 = arith.constant 2 : i64
    %cst_3 = arith.constant dense<64> : tensor<256xi32>
    %c16_i32 = arith.constant 16 : i32
    %c2_i32 = arith.constant 2 : i32
    %c32_i32 = arith.constant 32 : i32
    %c4_i32 = arith.constant 4 : i32
    %cst_4 = arith.constant dense<384> : tensor<256xi32>
    %c256_i32 = arith.constant 256 : i32
    %0 = tt.get_program_id x : i32
    %1 = arith.muli %0, %c256_i32 : i32
    %2 = tt.make_range {end = 256 : i32, start = 0 : i32} : tensor<256xi32>
    %3 = tt.splat %1 : i32 -> tensor<256xi32>
    %4 = arith.addi %3, %2 : tensor<256xi32>
    %5 = arith.cmpi slt, %4, %cst_4 : tensor<256xi32>
    %6 = arith.divsi %4, %cst_3 : tensor<256xi32>
    %7 = arith.divsi %1, %c32_i32 : i32
    %8 = arith.remsi %7, %c2_i32 : i32
    %9 = arith.divsi %1, %c4_i32 : i32
    %10 = arith.remsi %9, %c4_i32 : i32
    %11 = tt.make_tensor_ptr %arg0, [%c2_i64, %c4_i64], [%c4_i64, %c1_i64], [%8, %10] {order = array<i32: 1, 0>} : <tensor<2x4xi64>>
    %12 = tt.load %11 evictionPolicy = evict_last : !tt.ptr<tensor<2x4xi64>>
    %13 = tt.expand_dims %12 {axis = 0 : i32} : tensor<2x4xi64> -> tensor<1x2x4xi64>
    %14 = tt.expand_dims %13 {axis = 2 : i32} : tensor<1x2x4xi64> -> tensor<1x2x1x4xi64>
    %15 = tt.expand_dims %14 {axis = 4 : i32} : tensor<1x2x1x4xi64> -> tensor<1x2x1x4x1xi64>
    %16 = tt.broadcast %15 : tensor<1x2x1x4x1xi64> -> tensor<4x2x2x4x4xi64>
    %17 = tt.reshape %16 : tensor<4x2x2x4x4xi64> -> tensor<256xi64>
    %18 = arith.divsi %1, %c16_i32 : i32
    %19 = arith.remsi %18, %c2_i32 : i32
    %20 = arith.remsi %1, %c4_i32 : i32
    %21 = tt.make_tensor_ptr %arg0, [%c2_i64, %c4_i64], [%c4_i64, %c1_i64], [%19, %20] {order = array<i32: 1, 0>} : <tensor<2x4xi64>>
    %22 = tt.load %21 evictionPolicy = evict_last : !tt.ptr<tensor<2x4xi64>>
    %23 = tt.expand_dims %22 {axis = 0 : i32} : tensor<2x4xi64> -> tensor<1x2x4xi64>
    %24 = tt.expand_dims %23 {axis = 2 : i32} : tensor<1x2x4xi64> -> tensor<1x2x1x4xi64>
    %25 = tt.broadcast %24 : tensor<1x2x1x4xi64> -> tensor<8x2x4x4xi64>
    %26 = tt.reshape %25 : tensor<8x2x4x4xi64> -> tensor<256xi64>
    %27 = arith.addi %17, %cst_1 : tensor<256xi64>
    %28 = arith.cmpi slt, %17, %cst : tensor<256xi64>
    %29 = arith.select %28, %27, %17 : tensor<256xi1>, tensor<256xi64>
    %30 = arith.cmpi sge, %29, %cst : tensor<256xi64>
    %31 = arith.cmpi slt, %29, %cst_1 : tensor<256xi64>
    %32 = arith.andi %30, %31 : tensor<256xi1>
    %33 = arith.xori %5, %cst_2 : tensor<256xi1>
    %34 = arith.ori %32, %33 : tensor<256xi1>
    tt.assert %34, "index out of bounds: 0 <= tmp4 < 5" : tensor<256xi1>
    %35 = arith.addi %26, %cst_1 : tensor<256xi64>
    %36 = arith.cmpi slt, %26, %cst : tensor<256xi64>
    %37 = arith.select %36, %35, %26 : tensor<256xi1>, tensor<256xi64>
    %38 = arith.cmpi sge, %37, %cst : tensor<256xi64>
    %39 = arith.cmpi slt, %37, %cst_1 : tensor<256xi64>
    %40 = arith.andi %38, %39 : tensor<256xi1>
    %41 = arith.ori %40, %33 : tensor<256xi1>
    tt.assert %41, "index out of bounds: 0 <= tmp9 < 5" : tensor<256xi1>
    %42 = arith.muli %29, %cst_1 : tensor<256xi64>
    %43 = arith.addi %37, %42 : tensor<256xi64>
    %44 = arith.muli %6, %cst_0 : tensor<256xi32>
    %45 = arith.extsi %44 : tensor<256xi32> to tensor<256xi64>
    %46 = arith.addi %43, %45 : tensor<256xi64>
    %47 = tt.splat %arg1 : !tt.ptr<f16> -> tensor<256x!tt.ptr<f16>>
    %48 = tt.addptr %47, %46 : tensor<256x!tt.ptr<f16>>, tensor<256xi64>
    %49 = tt.load %48, %5 evictionPolicy = evict_last : tensor<256x!tt.ptr<f16>>
    %50 = tt.make_tensor_ptr %arg2, [%c384_i64], [%c1_i64], [%1] {order = array<i32: 0>} : <tensor<256xf16>>
    tt.store %50, %49 {boundaryCheck = array<i32: 0>} : !tt.ptr<tensor<256xf16>>
    tt.return
  }
}

{-#
  external_resources: {
    mlir_reproducer: {
      pipeline: "builtin.module(convert-triton-to-tritongpu{enable-source-remat=false num-ctas=1 num-warps=4 target=xpu threads-per-warp=32}, tritonintelgpu-coalesce, tritonintelgpu-remove-layout-conversions, tritonintelgpu-accelerate-matmul, tritonintelgpu-materialize-block-pointer, tritonintelgpu-remove-layout-conversions, tritonintelgpu-optimize-dot-operands, tritonintelgpu-pipeline{num-stages=1 split-barriers-scope=none}, tritonintelgpu-reduce-variable-liveness, tritongpu-fuse-nested-loops, canonicalize{  max-iterations=10 max-num-rewrites=-1 region-simplify=normal test-convergence=false top-down=true}, triton-licm, canonicalize{  max-iterations=10 max-num-rewrites=-1 region-simplify=normal test-convergence=false top-down=true}, tritongpu-combine-tensor-select-and-if, tritongpu-optimize-thread-locality, tritongpu-optimize-dot-operands{hoist-layout-conversion=true}, cse, tritongpu-prefetch, tritongpu-optimize-dot-operands{hoist-layout-conversion=true}, tritonintelgpu-remove-layout-conversions, tritonintelgpu-reduce-data-duplication, tritongpu-reorder-instructions, cse, symbol-dce, sccp, canonicalize{  max-iterations=10 max-num-rewrites=-1 region-simplify=normal test-convergence=false top-down=true}, arith-emulate-unsupported-floats{source-types={bf16} target-type=f32})",
      disable_threading: false,
      verify_each: true
    }
  }
#-}
/tmp/torchinductor_jovyan/ke/ckezo5lp275omoh5gyhtmuqbk7txr3j56bdd67qrripatlqbpr7f.py:18:0: error: Failures have been detected while processing an MLIR pass pipeline
/tmp/torchinductor_jovyan/ke/ckezo5lp275omoh5gyhtmuqbk7txr3j56bdd67qrripatlqbpr7f.py:18:0: note: Pipeline failed while executing [`TritonIntelGPUMaterializeBlockPointer` on 'builtin.module' operation]: reproducer generated at `std::errs, please share the reproducer above with Triton project.`
E0716 07:25:08.857000 3132999 .scripts_cache/pytorch/torch/_inductor/runtime/triton_heuristics.py:714] Triton compilation failed: triton_poi_fused_im2col_1
E0716 07:25:08.857000 3132999 .scripts_cache/pytorch/torch/_inductor/runtime/triton_heuristics.py:714] def triton_poi_fused_im2col_1(in_ptr0, in_ptr1, out_ptr0, xnumel, XBLOCK : tl.constexpr):
E0716 07:25:08.857000 3132999 .scripts_cache/pytorch/torch/_inductor/runtime/triton_heuristics.py:714]     xnumel = 384
E0716 07:25:08.857000 3132999 .scripts_cache/pytorch/torch/_inductor/runtime/triton_heuristics.py:714]     xoffset = tl.program_id(0) * XBLOCK
E0716 07:25:08.857000 3132999 .scripts_cache/pytorch/torch/_inductor/runtime/triton_heuristics.py:714]     xindex = xoffset + tl.arange(0, XBLOCK)[:]
E0716 07:25:08.857000 3132999 .scripts_cache/pytorch/torch/_inductor/runtime/triton_heuristics.py:714]     xmask = xindex < xnumel
E0716 07:25:08.857000 3132999 .scripts_cache/pytorch/torch/_inductor/runtime/triton_heuristics.py:714]     x1 = ((xindex // 4) % 4)
E0716 07:25:08.857000 3132999 .scripts_cache/pytorch/torch/_inductor/runtime/triton_heuristics.py:714]     x3 = ((xindex // 32) % 2)
E0716 07:25:08.857000 3132999 .scripts_cache/pytorch/torch/_inductor/runtime/triton_heuristics.py:714]     x0 = (xindex % 4)
E0716 07:25:08.857000 3132999 .scripts_cache/pytorch/torch/_inductor/runtime/triton_heuristics.py:714]     x2 = ((xindex // 16) % 2)
E0716 07:25:08.857000 3132999 .scripts_cache/pytorch/torch/_inductor/runtime/triton_heuristics.py:714]     x4 = xindex // 64
E0716 07:25:08.857000 3132999 .scripts_cache/pytorch/torch/_inductor/runtime/triton_heuristics.py:714]     x7 = xindex
E0716 07:25:08.857000 3132999 .scripts_cache/pytorch/torch/_inductor/runtime/triton_heuristics.py:714]     tmp0 = tl.reshape(tl.broadcast_to(tl.load(tl.make_block_ptr(in_ptr0, shape=[2, 4], strides=[4, 1], block_shape=[((2) * ((2) <= ((31 + XBLOCK) // 32)) + ((31 + XBLOCK) // 32) * (((31 + XBLOCK) // 32) < (2))), ((4) * ((4) <= ((3 + XBLOCK) // 4)) + ((3 + XBLOCK) // 4) * (((3 + XBLOCK) // 4) < (4)))], order=[1, 0], offsets=[((xoffset // 32) % 2), ((xoffset // 4) % 4)]), eviction_policy='evict_last')[None, :, None, :, None], [(63 + XBLOCK) // 64, ((2) * ((2) <= ((31 + XBLOCK) // 32)) + ((31 + XBLOCK) // 32) * (((31 + XBLOCK) // 32) < (2))), ((2) * ((2) <= ((15 + XBLOCK) // 16)) + ((15 + XBLOCK) // 16) * (((15 + XBLOCK) // 16) < (2))), ((4) * ((4) <= ((3 + XBLOCK) // 4)) + ((3 + XBLOCK) // 4) * (((3 + XBLOCK) // 4) < (4))), ((4) * ((4) <= (XBLOCK)) + (XBLOCK) * ((XBLOCK) < (4)))]), [XBLOCK])
E0716 07:25:08.857000 3132999 .scripts_cache/pytorch/torch/_inductor/runtime/triton_heuristics.py:714]     tmp6 = tl.reshape(tl.broadcast_to(tl.load(tl.make_block_ptr(in_ptr0, shape=[2, 4], strides=[4, 1], block_shape=[((2) * ((2) <= ((15 + XBLOCK) // 16)) + ((15 + XBLOCK) // 16) * (((15 + XBLOCK) // 16) < (2))), ((4) * ((4) <= (XBLOCK)) + (XBLOCK) * ((XBLOCK) < (4)))], order=[1, 0], offsets=[((xoffset // 16) % 2), (xoffset % 4)]), eviction_policy='evict_last')[None, :, None, :], [(31 + XBLOCK) // 32, ((2) * ((2) <= ((15 + XBLOCK) // 16)) + ((15 + XBLOCK) // 16) * (((15 + XBLOCK) // 16) < (2))), ((4) * ((4) <= ((3 + XBLOCK) // 4)) + ((3 + XBLOCK) // 4) * (((3 + XBLOCK) // 4) < (4))), ((4) * ((4) <= (XBLOCK)) + (XBLOCK) * ((XBLOCK) < (4)))]), [XBLOCK])
E0716 07:25:08.857000 3132999 .scripts_cache/pytorch/torch/_inductor/runtime/triton_heuristics.py:714]     tmp1 = tl.full([XBLOCK], 5, tl.int32)
E0716 07:25:08.857000 3132999 .scripts_cache/pytorch/torch/_inductor/runtime/triton_heuristics.py:714]     tmp2 = tmp0 + tmp1
E0716 07:25:08.857000 3132999 .scripts_cache/pytorch/torch/_inductor/runtime/triton_heuristics.py:714]     tmp3 = tmp0 < 0
E0716 07:25:08.857000 3132999 .scripts_cache/pytorch/torch/_inductor/runtime/triton_heuristics.py:714]     tmp4 = tl.where(tmp3, tmp2, tmp0)
E0716 07:25:08.857000 3132999 .scripts_cache/pytorch/torch/_inductor/runtime/triton_heuristics.py:714]     tl.device_assert(((0 <= tmp4) & (tmp4 < 5)) | ~(xmask), "index out of bounds: 0 <= tmp4 < 5")
E0716 07:25:08.857000 3132999 .scripts_cache/pytorch/torch/_inductor/runtime/triton_heuristics.py:714]     tmp7 = tmp6 + tmp1
E0716 07:25:08.857000 3132999 .scripts_cache/pytorch/torch/_inductor/runtime/triton_heuristics.py:714]     tmp8 = tmp6 < 0
E0716 07:25:08.857000 3132999 .scripts_cache/pytorch/torch/_inductor/runtime/triton_heuristics.py:714]     tmp9 = tl.where(tmp8, tmp7, tmp6)
E0716 07:25:08.857000 3132999 .scripts_cache/pytorch/torch/_inductor/runtime/triton_heuristics.py:714]     tl.device_assert(((0 <= tmp9) & (tmp9 < 5)) | ~(xmask), "index out of bounds: 0 <= tmp9 < 5")
E0716 07:25:08.857000 3132999 .scripts_cache/pytorch/torch/_inductor/runtime/triton_heuristics.py:714]     tmp11 = tl.load(in_ptr1 + (tmp9 + 5*tmp4 + 25*x4), xmask, eviction_policy='evict_last').to(tl.float32)
E0716 07:25:08.857000 3132999 .scripts_cache/pytorch/torch/_inductor/runtime/triton_heuristics.py:714]     tl.store(tl.make_block_ptr(out_ptr0, shape=[384], strides=[1], block_shape=[XBLOCK], order=[0], offsets=[xoffset]), tl.broadcast_to(tmp11, [XBLOCK]).to(tl.float16), boundary_check=[0])
E0716 07:25:08.857000 3132999 .scripts_cache/pytorch/torch/_inductor/runtime/triton_heuristics.py:714] 
E0716 07:25:08.857000 3132999 .scripts_cache/pytorch/torch/_inductor/runtime/triton_heuristics.py:714] metadata: {'signature': {'in_ptr0': '*i64', 'in_ptr1': '*fp16', 'out_ptr0': '*fp16', 'xnumel': 'i32', 'XBLOCK': 'constexpr'}, 'device': 0, 'constants': {'XBLOCK': 256}, 'configs': [{(0,): [['tt.divisibility', 16]], (1,): [['tt.divisibility', 16]], (2,): [['tt.divisibility', 16]], (3,): [['tt.divisibility', 16]]}], 'device_type': 'xpu', 'num_warps': 4, 'num_stages': 1, 'debug': True, 'cc': {'architecture': 13136561920, 'device_id': 3029, 'driver_version': '1.6.33276+21', 'gpu_eu_count': 512, 'gpu_subslice_count': 64, 'has_atomic64': True, 'has_bfloat16_conversions': True, 'has_fp16': True, 'has_fp64': True, 'has_subgroup_2d_block_io': True, 'has_subgroup_matrix_multiply_accumulate': True, 'has_subgroup_matrix_multiply_accumulate_tensor_float32': False, 'max_compute_units': 512, 'max_num_sub_groups': 64, 'max_work_group_size': 1024, 'name': 'Intel(R) Data Center GPU Max 1550', 'platform_name': 'Intel(R) oneAPI Unified Runtime over Level-Zero', 'sub_group_sizes': [16, 32], 'total_memory': 68702699520, 'type': 'gpu', 'vendor': 'Intel(R) Corporation', 'version': '12.60.7'}}
E0716 07:25:08.857000 3132999 .scripts_cache/pytorch/torch/_inductor/runtime/triton_heuristics.py:714] Traceback (most recent call last):
E0716 07:25:08.857000 3132999 .scripts_cache/pytorch/torch/_inductor/runtime/triton_heuristics.py:714]   File "/home/jovyan/intel-xpu-backend-for-triton/.scripts_cache/pytorch/torch/_inductor/runtime/triton_heuristics.py", line 712, in _precompile_config
E0716 07:25:08.857000 3132999 .scripts_cache/pytorch/torch/_inductor/runtime/triton_heuristics.py:714]     binary = triton.compile(*compile_args, **compile_kwargs)
E0716 07:25:08.857000 3132999 .scripts_cache/pytorch/torch/_inductor/runtime/triton_heuristics.py:714]   File "/home/jovyan/intel-xpu-backend-for-triton/python/triton/compiler/compiler.py", line 326, in compile
E0716 07:25:08.857000 3132999 .scripts_cache/pytorch/torch/_inductor/runtime/triton_heuristics.py:714]     next_module = compile_ir(module, metadata)
E0716 07:25:08.857000 3132999 .scripts_cache/pytorch/torch/_inductor/runtime/triton_heuristics.py:714]   File "/home/jovyan/intel-xpu-backend-for-triton/python/triton/backends/intel/compiler.py", line 431, in <lambda>
E0716 07:25:08.857000 3132999 .scripts_cache/pytorch/torch/_inductor/runtime/triton_heuristics.py:714]     stages["ttgir"] = lambda src, metadata: self.make_ttgir(src, metadata, options, self.properties)
E0716 07:25:08.857000 3132999 .scripts_cache/pytorch/torch/_inductor/runtime/triton_heuristics.py:714]   File "/home/jovyan/intel-xpu-backend-for-triton/python/triton/backends/intel/compiler.py", line 287, in make_ttgir
E0716 07:25:08.857000 3132999 .scripts_cache/pytorch/torch/_inductor/runtime/triton_heuristics.py:714]     pm.run(mod)
E0716 07:25:08.857000 3132999 .scripts_cache/pytorch/torch/_inductor/runtime/triton_heuristics.py:714] RuntimeError: PassManager::run failed
Traceback (most recent call last):
  File "/home/jovyan/intel-xpu-backend-for-triton/issue_4725.py", line 84, in <module>
    triton_poi_fused_im2col_1 = async_compile.triton('triton_poi_fused_im2col_1', '''
  File "/home/jovyan/intel-xpu-backend-for-triton/.scripts_cache/pytorch/torch/_inductor/async_compile.py", line 401, in triton
    kernel.precompile(
  File "/home/jovyan/intel-xpu-backend-for-triton/.scripts_cache/pytorch/torch/_inductor/runtime/triton_heuristics.py", line 407, in precompile
    self._precompile_worker()
  File "/home/jovyan/intel-xpu-backend-for-triton/.scripts_cache/pytorch/torch/_inductor/runtime/triton_heuristics.py", line 429, in _precompile_worker
    compile_results.append(self._precompile_config(c))
  File "/home/jovyan/intel-xpu-backend-for-triton/.scripts_cache/pytorch/torch/_inductor/runtime/triton_heuristics.py", line 712, in _precompile_config
    binary = triton.compile(*compile_args, **compile_kwargs)
  File "/home/jovyan/intel-xpu-backend-for-triton/python/triton/compiler/compiler.py", line 326, in compile
    next_module = compile_ir(module, metadata)
  File "/home/jovyan/intel-xpu-backend-for-triton/python/triton/backends/intel/compiler.py", line 431, in <lambda>
    stages["ttgir"] = lambda src, metadata: self.make_ttgir(src, metadata, options, self.properties)
  File "/home/jovyan/intel-xpu-backend-for-triton/python/triton/backends/intel/compiler.py", line 287, in make_ttgir
    pm.run(mod)
RuntimeError: PassManager::run failed
```
</details>

--- a/test/TritonIntelGPU/materialize-block-pointer.mlir
+++ b/test/TritonIntelGPU/materialize-block-pointer.mlir
@@ -185,3 +185,26 @@ module attributes {"ttg.num-ctas" = 1 : i32, "ttg.num-warps" = 32 : i32, "ttg.th
     tt.return
   }
 }
+
+// -----
+
+// COM: Ensure i64 element type is supported in materialize block pointer.
+
+#dpas = #ttig.dpas<{repeatCount = 8, systolicDepth = 8, executionSize = 16, opsPerChan = 2, threadsPerWarp = 16, warpsPerCTA = [4, 2], repCluster = [1, 1], A = [8, 16], B = [16, 16], C = [8, 16]}>
+#dot_a = #ttg.dot_op<{opIdx = 0, parent = #dpas, kWidth = 1}>
+module attributes {"ttg.num-ctas" = 1 : i32, ttg.target = "xpu", "ttg.threads-per-warp" = 16 : i32, ttig.support_sg_2d_block} {
+  // CHECK-LABEL: tt.func public @materialize_block_pointer(
+  tt.func public @materialize_block_pointer(%arg0: !tt.ptr<i64> {tt.divisibility = 16 : i32}, %pitch: i64 {tt.divisibility = 16 : i32}) {
+    %c0_i32 = arith.constant 0 : i32
+    %c0_i64 = arith.constant 0 : i64
+    %c1_i64 = arith.constant 1 : i64
+
+    // CHECK: tt.load {{.*}} {ttig.block_io = "row_major"}
+    %0 = tt.make_tensor_ptr %arg0, [%c0_i64, %c0_i64], [%pitch, %c1_i64], [%c0_i32, %c0_i32] {order = array<i32: 1, 0>} : <tensor<64x32xi64, #dot_a>>
+    %1 = tt.load %0 : !tt.ptr<tensor<64x32xi64, #dot_a>>
+    // CHECK: tt.store {{.*}} {ttig.block_io = "row_major"}
+    tt.store %0, %1 : !tt.ptr<tensor<64x32xi64, #dot_a>>
+
+    tt.return
+  }
+}

